src/ActivityPub/ActorHandle.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\ActivityPub;
+
+class ActorHandle
+{
+    public const HANDLE_PATTERN = '/^(?P<prefix>[@!])?(?P<user>[\w\-\.]+)@(?P<host>[\w\.\-]+)(?P<port>:[\d]+)?$/';
```

Comment:
I hope this works well 🤗. Maybe you can add a unit test to validate this regex on various strings?

Reply:
I plucked this regex from the webfinger part and modify a bit to also recognize and extract additional prefix part, so it should be fine for the time being

a bit of unit tests to check this wouldn't hurt but then I'll need those samples to make it, also it's out of scope for this patch, feel free to file a patch

Reply:
well in any case I added a small unit tests for this, if you have any more interesting handle samples feel free to add it there